### ISSUE TITLE
int should be converted to string when sending over the API

### ISSUE THREAD

**Issue by alexeyinkin** [stvndelucis/firestore_api_parser#7]
Currently an `int` is converted to `{"integerValue": 123}`
However, an `int` comes from the server as `{"integerValue": "123"}`.
So it should be converted back to string when sending it to the server for symmetry.

**Comment by stvndelucis** [comment]
Hello! 
Thanks for the comment, I'll correct it.

**Comment by stvndelucis** [comment]
@alexeyinkin I've noticed that as long as you send int data with the **integerValue** key, firestore understands and stores it as int. So I don't think this extra action is necessary.

What do you think of it?

**Comment by alexeyinkin** [comment]
That was important for my case.

My Dart app writes a JSON on disk. Then Terraform picks that JSON and diffs it against the Firestore. If any difference is found, it updates the database.

In my case, the type difference alone was causing a non-empty diff. Database had "123", while the local file had 123. So it was running a lot of updates in vain.

The rule of thumb is that each type has its canonical representation. If the database returns it in some format, the exactly same format must be sent to it even if the database is smart enough to accept other formats. This prevents cases like mine, and you never know what cases users have.

Same problems are with:
- Timestamp. Firestore expects milliseconds only while DateTime.toString() produces microseconds. Microseconds cause false diff.
- Map. In Firestore, empty map does not have "fields". Empty "fields" causes false diff.
- Array. In Firestore, empty array does not have "values". Empty "values" causes false diff.

Anyway, I have already created and published my own package that addresses all the issues I filed here: https://pub.dev/packages/firestore_api_converter

You are free to use for yourself it or copy any solutions from there to this package.
